### PR TITLE
implemented 'per_value' train splitting

### PR DESCRIPTION
**Pull request by iovsn**
Implemented 'per_value' splitting, where class ids get assigned to the same subclasses across multiple eopatches.

### CODE REVIEW

File: ml_tools/eolearn/ml_tools/train_test_split.py [hunk 1]
```diff
@@ -28,13 +44,13 @@ class TrainTestSplitTask(EOTask):
...
 
-    def __init__(self, feature, bins, split_type='per_class', no_data_value=None):
+    def __init__(self, feature, bins, split_type=TrainTestSplitType.PER_CLASS, no_data_value=None):
         """
         :param feature: The input feature out of which to generate the train mask.
         :type feature: (FeatureType, feature_name, new_name)
         :param bins: Cumulative probabilities of all value classes or a single float, representing a fraction.
         :type bins: a float or list of floats
-        :param split_type: Valye split type, either 'per_class' or 'random'.
+        :param split_type: Valye split type, either 'per_class' or 'per_pixel'.
```
[the remark this answers is not in the document]
Comment (devisperessutti) [reply ml_tools/eolearn/ml_tools/train_test_split.py]: typo in value

File: ml_tools/eolearn/ml_tools/train_test_split.py [hunk 2]
```diff
@@ -64,28 +77,25 @@ def execute(self, eopatch, *, seed=None):
...
-        np.random.seed(seed)
+        if self.split_type in [TrainTestSplitType.PER_CLASS, TrainTestSplitType.PER_PIXEL]:
+            np.random.seed(seed)
 
         ftype, fname, new_name = self.feature
         data = eopatch[(ftype, fname)]
 
-        if self.split_type == 'per_class':
+        if self.split_type == TrainTestSplitType.PER_PIXEL:
+            rands = np.random.rand(*data.shape)
```
Comment (devisperessutti) [inline ml_tools/eolearn/ml_tools/train_test_split.py]: as this task is currently designed to work with `MASK_TIMELESS` type, I would add to the `_parse_feature` method on line 58 the argument `allowed_feature_types={FeatureType.MASK_TIMELESS}`, to avoid mis-behaviours (e.g. if `MASK` is specified instead)

File: ml_tools/eolearn/ml_tools/train_test_split.py [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
     The group of pixels is defined by an input feature (i.e. MASK_TIMELESS with polygon ids, or connected component id,
-    or similar that groups pixels with similar properties). By default pixels get assigned to a subset 'per_class'. If
-    split_type is set to 'random', then pixels get randomly assigned to a subset, regardless of their class.
+    or similar that groups pixels with similar properties). By default pixels get assigned to a subset
+    :attr:`PER_CLASS<eolearn.ml_tools.train_test_split.TrainTestSplitType.PER_CLASS>`, where pixels of the same class
+    will be assigned to the same subset. On the other hand if split_type is set to
+    :attr:`PER_PIXEL<eolearn.ml_tools.train_test_split.TrainTestSplitType.PER_PIXEL>`, each pixel gets randomly
+    assigned to a subset regardless of its class. The third option is splitting
+    :attr:`PER_VALUE<eolearn.ml_tools.train_test_split.TrainTestSplitType.PER_VALUE>`, where each class is assigned to
+    a certain subset consistently across eopatches. In other words, if a polyogn lies on multiple eopatches, it will
```
[the remark this answers is not in the document]
Comment (iovsn) [reply ml_tools/eolearn/ml_tools/train_test_split.py]: What about 'pixel' instead of 'value'? It's used in the text multiple times already. Either 'value' or 'pixel', it would maybe be good to be consistent and use just one of them.

[the remark this answers is not in the document]
Comment (iovsn) [reply ml_tools/eolearn/ml_tools/train_test_split.py]: Refactored the whole docstring more generally.

File: ml_tools/eolearn/ml_tools/train_test_split.py [hunk 4]
```diff
@@ -28,13 +44,13 @@ class TrainTestSplitTask(EOTask):
     have a value representing the train, test and/or validation set.
     """
 
-    def __init__(self, feature, bins, split_type='per_class', no_data_value=None):
+    def __init__(self, feature, bins, split_type=TrainTestSplitType.PER_CLASS, no_data_value=None):
```
[the remark this answers is not in the document]
Comment (iovsn) [reply ml_tools/eolearn/ml_tools/train_test_split.py]: Done.

File: ml_tools/eolearn/ml_tools/train_test_split.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (iovsn) [reply ml_tools/eolearn/ml_tools/train_test_split.py]: Done.

File: ml_tools/eolearn/ml_tools/train_test_split.py (hunk 2 above)
Comment (iovsn) [reply ml_tools/eolearn/ml_tools/train_test_split.py]: That is not true. It should handle almost any feature type, probably with a few exceptions. In the unit tests there are cases for MASK_TIMELESS, LABEL_TIMELESS and DATA wich is tested using 4D arrays.
